packages/runtime/container-runtime/src/gc/gcConfigs.ts

```diff
@@ -107,32 +102,24 @@ export function generateGCConfigs(
 	);
 
 	/**
-	 * Whether sweep should run or not. This refers to whether Tombstones should fail on load and whether
-	 * sweep-ready nodes should be deleted.
+	 * Whether sweep should is enabled for this session or not. This refers to whether Tombstones should fail on load
+	 * and whether sweep-ready nodes should be deleted.
 	 *
-	 * Assuming overall GC is enabled and tombstoneTimeout is provided, the following conditions have to be met to run sweep:
+	 * Assuming overall GC is allowed and tombstoneTimeout is provided, the following conditions have to be met to run sweep:
 	 *
 	 * 1. Sweep should be allowed in this container.
 	 * 2. Sweep should be enabled for this session, optionally restricted to attachment blobs only.
 	 *
 	 * These conditions can be overridden via the RunSweep feature flag.
 	 */
 	const sweepEnabled: boolean =
-		!gcEnabled || tombstoneTimeoutMs === undefined
+		!gcAllowed || tombstoneTimeoutMs === undefined
 			? false
 			: sweepAllowed && createParams.gcOptions.enableGCSweep === true;
-	const disableDatastoreSweep =
-		mc.config.getBoolean(disableDatastoreSweepKey) === true ||
-		createParams.gcOptions[gcDisableDataStoreSweepOptionName] === true;
-	const shouldRunSweep: IGarbageCollectorConfigs["shouldRunSweep"] = sweepEnabled
-		? disableDatastoreSweep
-			? "ONLY_BLOBS"
-			: "YES"
-		: "NO";
 
 	// If we aren't running sweep, also disable AutoRecovery which also emits the GC op.
 	// This gives a simple control surface for compability concerns around introducing the new op.
-	const tombstoneAutorecoveryEnabled = shouldRunSweep !== "NO";
+	const tombstoneAutorecoveryEnabled = sweepEnabled;
```

Comment:
if auto-recovery now goes hand in hand with sweep (according to this line of code); is there any value in keeping this `tombstoneAutorecoveryEnabled` separate? May be I am unclear on how this helps on the compat concerns!

AFAIU, auto-recovery makes GC less destructive; so keeping it on always, is only make Sweep more safe (and conservative)! isn't it?

Reply:
You are right. The option to disabled it should be removed. The logic to disable it is directly in the usage and not here, so I missed it. Let me do it separately so as to not make this PR bigger than it already is :-).